### PR TITLE
PR checker and updater for Jira missing stuff

### PR DESCRIPTION
**Pull request by AlexisErazoGlobant**
### Purpose

Sometimes authors forget to add the ticket number to the title or/and add a Jira link to the description. So I'm adding the GitHub Action to check if the Jira ticket number is missing in the title or in the description. This updates the title with the ticket number and adds a comment with the link redirecting to the Jira ticket.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

GitHub Action to check if the Jira ticket number is missing in the title or in the description. This updates the title with the ticket number and adds a comment with the link redirecting to the Jira ticket.

### Reviewers

@sm6srw

### CODE REVIEW

**Comment by QilongTang** [comment]
Is the title of this PR going to be updated with Jira ticket as well? :)

**Comment by AlexisErazoGlobant** [comment]
Hi @QilongTang, it's supposed to be like that, but I think we have to wait to finish the validations to see the result. 😅

**Review by sm6srw — COMMENTED** [review]
I did approve the new action to run. But it failed miserable. @avidit Is it something more I need to do?

**Comment by AlexisErazoGlobant** [comment]
@sm6srw remember the error that I had in my private repo? It seems to be the same, I solved doing this in GitHub Configurations.
posible-error-solution

**Comment by sm6srw** [comment]
Yes, those are the permissions we currently have,.

**Comment by AlexisErazoGlobant** [comment]
Can u re-run the job? Enabling the debug logging?

File: .github/workflows/pr_jira_check.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  title:
+    name: Missing Jira ticket check
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Check for ticket
+        uses: neofinancial/ticket-check-action@v1
```
Comment (sm6srw) [inline .github/workflows/pr_jira_check.yml:15]: I see warnings about that we try to use deprecated versions of node.js (12). Looks like the latest version of this is v2. Can you try to use v2 instead?

Comment (AlexisErazoGlobant) [reply .github/workflows/pr_jira_check.yml:15]: Let me try on private first and I let you know.

Comment (AlexisErazoGlobant) [reply .github/workflows/pr_jira_check.yml]: @sm6srw I did it on private and it works. PR updated.

Comment (AlexisErazoGlobant) [reply .github/workflows/pr_jira_check.yml]: It fails again, but I don't catch why :( maybe is the name of the branch, or idk 
image

File: .github/workflows/pr_jira_check.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: PR checker and updater for Jira missing stuff
```
Comment (sm6srw) [inline .github/workflows/pr_jira_check.yml:1]: I think we should change this to `Jira ticket check`

Comment (AlexisErazoGlobant) [reply .github/workflows/pr_jira_check.yml]: Updated!

**Comment by avidit** [comment]
> I did approve the new action to run. But it failed miserable. @avidit Is it something more I need to do?

When an action id triggered in the context of PR from a fork, the `secrets.GITHUB_TOKEN` has read only scopes. We can merge it to master and wait for the next PR. 
Then we can see if there might are issues in the action itself. 

Regarding Node.js 16 deprecation, the action itself needs to be updated. @AlexisErazoGlobant Can we find alternative action in the marketplace?

**Comment by AlexisErazoGlobant** [comment]
@avidit Let me check for it, previously I found another one, but it seems it's made for its own company and its own Jira links, we could check together if you guys want.

**Pull request merged**